pennylane/beta/plugins/default_tensor.py

```diff
@@ -503,13 +522,13 @@ def _ev_exact(self, obs_nodes, wires):
         meas_wires = []
         # For wires which are measured, add edges between
         # the ket node, the observable nodes, and the bra node
-        for obs_node, obs_wires in zip(obs_nodes, wires):
-            meas_wires.extend(obs_wires)
-            for idx, w in enumerate(obs_wires):
+        for obs_node, obs_registers in zip(obs_nodes, registers):
+            meas_wires.extend(obs_registers)
+            for idx, w in enumerate(obs_registers):
                 # Use convention that the indices of a tensor are ordered like
                 # [output_idx1, output_idx2, ..., input_idx1, input_idx2, ...]
                 output_idx = idx
-                input_idx = len(obs_wires) + idx
+                input_idx = len(obs_registers) + idx
                 tn.connect(obs_node[input_idx], ket[w])  # A|psi>
                 tn.connect(bra[w], obs_node[output_idx])  # <psi|A
```

Comment:
This is another place where things *might* break if `registers` is not a consecutive list, starting from 0. You can see there is some manual indexing going on, with the implicit assumption (in `enumerate`) that things start with 0